src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -142,11 +145,15 @@ def _allocate_slot(self, cud):
     #
     def _release_slot(self, slots):
         '''
-        As the opposite to `_allocate_slot()`, this method will be invoked if
-        a unit does no longer need the resources previously allocation above.
+        This method is envoked when a unit does not need the resources that it
+        had acquired. 
+
+        The input passed are the resource slots a unit had acquired when 
+        `_allocate_slots()` was called.
         '''
 
-        # reflect the request in the nodelist state.
+        # reflect the request in the nodelist state and change the state of the
+        # slots from BUSY to FREE.
```

Comment:
It adds the state transition.Otherwise the question is what is the state of the slots before? I will have to search all the code to find what is going on. When looking in a method or function the comments should give all the information to understand the method/function without having to understand the class and the base class.

Reply:
Did you actually check if this is the case? :-P  Because, the method actually just sets the given values.  You may argue that is not what it should do, but right now your comment does not reflect what the code does.  Wrong comments are worse than no comments...